src/tests/quantuminspire/qiskit/test_backend_qx.py

```diff
@@ -258,13 +259,22 @@ def test_get_experiment_results_multiple_single_shots(self):
         self.assertEqual(one_shot_results['0x2'], 4003)
         self.assertEqual(one_shot_results['0x3'], 984)
 
-    def test_validate_negative_shot_count(self):
+    def test_validate_shot_count_zero(self):
         simulator = QuantumInspireBackend(Mock(), Mock())
+        job_dict = self._basic_qobj_dictionary
+        job_dict['config']['shots'] = 1
```

Comment:
Oke, this is a bit odd, but the line:
`job = qiskit.qobj.QasmQobj.from_dict(job_dict)`
throws an exeption when the number of shots < 1, this has been changed in qiskit, more strict validation of their schemas.
Because the actual unit test is in the last line, where our own validation function in method run should fail when shots < 1, what I did is creating the job with a valid number of shots (1) to keep qiskit happy and then resetting it to 0 before we test that we throw an exception when shots < 1.

Reply:
I've added extra comments for this test case.